lib/security/JwsToken.ts

```diff
@@ -13,6 +13,48 @@ type VerifySignatureDelegate = (signedContent: string, signature: string, jwk: P
  * This class hides the JOSE and crypto library dependencies to allow support for additional crypto algorithms.
  */
 export default class JwsToken extends JoseToken {
+
+  // used for verification if a JSON Serialized JWS was given
+  private readonly protected: string | undefined;
+  private header: {[key: string]: any} | undefined;
+  private readonly signature: string | undefined;
+  private readonly isFlattenedJSONSerialized: boolean;
+
+  constructor (content: string | object, protected cryptoFactory: CryptoFactory) {
+    super(content, cryptoFactory);
+    // this.content is set to content, which may be the plaintext, or a compact JWS.
```

Comment:
Design: thanks for explicitly calling out the morphing behavior of `this.content` (this info wasn't there for JweToken), not sure if I like it because this is often a bad practice. In this case, the meaning of content changes depending on what is passed. I'd rather have this content to be of the same type (`string | object`), or have an additional property to the class if necessary.